package/src/components/Message/Message.tsx

```diff
@@ -752,12 +756,19 @@ const areEqual = <StreamChatGenerics extends DefaultStreamChatGenerics = Default
     (Array.isArray(prevMessageAttachments) &&
       Array.isArray(nextMessageAttachments) &&
       prevMessageAttachments.length === nextMessageAttachments.length &&
-      prevMessageAttachments.every((attachment, index) =>
-        attachment.type === 'image'
-          ? attachment.image_url === nextMessageAttachments[index].image_url &&
-            attachment.thumb_url === nextMessageAttachments[index].thumb_url
-          : attachment.type === nextMessageAttachments[index].type,
-      )) ||
+      prevMessageAttachments.every((attachment, index) => {
+        const attachmentKeysEqual =
+          attachment.type === 'image'
+            ? attachment.image_url === nextMessageAttachments[index].image_url &&
+              attachment.thumb_url === nextMessageAttachments[index].thumb_url
+            : attachment.type === nextMessageAttachments[index].type;
+
+        const customIsAttachmentEqual =
+          isAttachmentEqual &&
+          isAttachmentEqual(prevMessageAttachments[index], nextMessageAttachments[index]);
+
+        return attachmentKeysEqual && customIsAttachmentEqual;
```

Comment:
This condition needs some rethinking. In case `isAttachmentEqual` is not provided, then ideally the value of `customIsAttachmentEqual` should be true, since we only need to rely on `attachmentKeysEqual`

Reply:
But with current implementation, this will always return false if isAttachmentEqual prop function is not provided by customer.

Reply:
Please check the other cases as well